docs/handbook/sunray.md

```diff
@@ -0,0 +1,359 @@
+# pkg change-facet facet.version-lock.gnome/gnome-session=false
+# pkg change-facet facet.version-lock.system/display-manager/gdm=false
+# pkg change-facet version-lock.library/desktop/libgweather=false
+# pkg change-facet version-lock.gnome/window-manager/metacity=false
+```
+
+## install from current Hipster (2019)
+
+If you install new Hipster it is necessary to have the GNOME packages before (or after) obsoleting
+
+The last know version of OI GNOME packages are:
+
+    libgweather@2.30.3
+    gnome-session@2.32.1
+    gnome-panel@2.32.1
+    metacity@2.30.3
+    gdm@2.30.7
+    libgnomekbd@2.30.0
+    gnome-settings-daemon@2.32.1
+
+These are still in hipster repo, but are obsoleted empty metapackages.
+
+[here](http://pkg.toc.de/sunray/)  can you found the old GNOME packages with newer release date, so that these should be able to install it on current Hipster.
+For that add the publisher in that kind, that it can override packages of publisher openindiana.org
+
+```shell
+# pkg set-publisher --search-before=openindiana.org -g http://pkg.toc.de/sunray sunray
+# pkg set-publisher --non-sticky openindiana.org
+```
+
+and install the all packages from publisher sunray.
+
+
+**The version locks have to release anyway.**
+
+#### Xscreensaver
+
+latest Hipster delivers Xscreensaver only in 64bit. The PAM module of SunRay are only shipped in 32bit.
+So we need a Xscreensaver package with 32bit bins (recompile with enabled 32bit or take it from [here](http://pkg.toc.de/sunray) ).
+
+# Login Screen (gdm-greeter) won't reapear after logout
+
+If after logout the gdm-login not reappears, helps the following script installed at `/opt/SUNWut/lib/gdm/revivesrsession.py` and called by the helper script `/etc/opt/SUNWut/gdm/SunRayPostSession/helpers/revivesession`
+
+```
+/opt/SUNWut/lib/gdm/revivesrsession.py &
+```
+
+```python
+#!/usr/bin/env python
+
+'''
+ Reconnect SunRay X-session: some times after logout, the gdm will not start login screen on the X-session and the DTU remains in state 26D
+ INSTALL: - cp to /opt/SUNWut/lib/gdm/revivesrsession.py
+          - create script /etc/opt/SUNWut/gdm/SunRayPostSession/helpers/revivesession:
+            /opt/SUNWut/lib/gdm/revivesrsession.py &
+'''
+
+import subprocess as sp
+import re
+import pdb
+import time
+import logging
+
+logformat = "%(asctime)s %(levelname)s:%(message)s"
+logging.basicConfig(format = logformat)
+logger = logging.getLogger(__name__)
+
+logger.setLevel(logging.DEBUG)
+
+time.sleep(10)
+
+pid = sp.Popen(['pgrep','gdm-binary'], stdout=sp.PIPE).stdout.readline().strip()
+logger.debug("GDM pid %s ", pid)
+dpl = [ p.split()[2].strip(':')
+    for p in sp.Popen(['ptree', pid], stdout=sp.PIPE).stdout.readlines() if '/opt/SUNWut/lib/Xnewt' in p ]
+logger.debug("Xnewt pid {}".format( dpl))
+
+# error -4 gdm-simple-slave not started for Display, no UT sessions
+for sess in sp.Popen(['pfexec','/opt/SUNWut/sbin/utsession','-px'], stdout=sp.PIPE).stdout.readlines():
```

Comment:
Does one need to have `pfexec` setup first? If so what privileges?

Reply:
You are right. My first thought was it is running by user gdm, but it is running as root. So it is not necessary and will remove pfexec